lib/qam.pm

```diff
@@ -86,12 +86,12 @@ sub add_extra_customer_repositories {
     my @repo_list = (
         {cond => '=12-SP2', name => '12-SP2-LTSS-ERICSSON-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP2-LTSS-ERICSSON/$arch/update/"},
         {cond => '=12-SP3', name => '12-SP3-LTSS-TERADATA-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP3-LTSS-TERADATA/$arch/update/"},
-        {cond => '=15-SP3', name => '15-SP3-ERICSSON-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP3-ERICSSON/$arch/update/"},
-        {cond => '=15-SP4', name => '15-SP4-ERICSSON-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP4-ERICSSON/$arch/update/"}
+        {cond => '=15-SP3', name => '15-SP3-ERICSSON-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP3-ERICSSON/$arch/update/", arch_only => 'x86_64'},
+        {cond => '=15-SP4', name => '15-SP4-ERICSSON-Updates', uri => "http://dist.suse.de/ibs/SUSE/Updates/SLE-Product-SLES/15-SP4-ERICSSON/$arch/update/", arch_only => 'x86_64'}
     );
 
     for my $repo (@repo_list) {
-        add_repo_if_not_present($repo->{uri}, $repo->{name}) if is_sle($repo->{cond});
+        add_repo_if_not_present($repo->{uri}, $repo->{name}) if is_sle($repo->{cond}) && defined($repo->{arch_only}) eq 'x86_64';
```

Comment:
I don't think the new condition is correct. [Perl docs](https://perldoc.perl.org/5.30.0/functions/defined) say that `defined()` returns boolean value.

Reply:
You're right https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17208